posydon/unit_tests/popsyn/test_star_formation_history.py

```diff
@@ -0,0 +1,1059 @@
+    
+    def test_chruslinska_model(self, monkeypatch):
+        """Test that get_SFH_model returns Chruslinska21 instance."""
+        # Mock the methods needed for initialization
+        def mock_load_data(self):
+            # Minimal setup to make initialization work
+            self.FOH_bins = np.linspace(5.3, 9.7, 10)
+            self.dFOH = self.FOH_bins[1] - self.FOH_bins[0]
+            self.Z = np.array([0.001, 0.01, 0.02])
+            self.redshifts = np.array([0.0, 1.0, 2.0])
+            self.SFR_data = np.ones((3, 10))
+            
+        def mock_load_redshift(self, verbose=False):
+            # Return mock time, redshift, deltaT
+            return (np.array([1e9, 2e9, 3e9]), 
+                   np.array([0.0, 1.0, 2.0]), 
+                   np.array([1e9, 1e9, 1e9]))
+            
+        def mock_load_raw(self):
+            # Return mock data matrix
+            return np.ones((3, 10)) * 1e6
+        
+        # Patch the methods
+        monkeypatch.setattr(Chruslinska21, "_load_chruslinska_data", mock_load_data)
+        monkeypatch.setattr(Chruslinska21, "_load_redshift_data", mock_load_redshift)
+        monkeypatch.setattr(Chruslinska21, "_load_raw_data", mock_load_raw)
+        
+        # Test the model creation
+        model_dict = {
+            "SFR": "Chruslinska+21",
+            "sub_model": "test",
+            "Z_solar_scaling": "Asplund09",
+            "Z_max": 0.03
+        }
+        model = get_SFH_model(model_dict)
+        assert isinstance(model, Chruslinska21)
+    
+    def test_zavala_model(self, monkeypatch):
+        """Test that get_SFH_model returns Zavala21 instance."""
+        # Mock the data loading method
+        def mock_load_data(self):
+            # Set required attributes directly
+            self.redshifts = np.array([0.0, 1.0, 2.0])
+            if self.sub_model == "min":
+                self.SFR_data = np.array([0.1, 0.08, 0.06])
+            else:
+                self.SFR_data = np.array([0.2, 0.16, 0.12])
+        
+        # Patch the data loading method
+        monkeypatch.setattr(Zavala21, "_load_zavala_data", mock_load_data)
+        
+        # Test for min model
+        model_dict = {
+            "SFR": "Zavala+21", 
+            "sub_model": "min",
+            "sigma": 0.5,
+            "Z_max": 0.03
+        }
+        model = get_SFH_model(model_dict)
+        assert isinstance(model, Zavala21)
+        assert model.sub_model == "min"
+        
+        # Test for max model
+        model_dict = {
+            "SFR": "Zavala+21", 
+            "sub_model": "max",
+            "sigma": 0.5,
+            "Z_max": 0.03
+        }
+        model = get_SFH_model(model_dict)
+        assert isinstance(model, Zavala21)
+        assert model.sub_model == "max"
+    
+    def test_invalid_model(self):
+        """Test that get_SFH_model raises an error for an invalid model."""
+        
+        model_dict = {"SFR": "InvalidModel", "sigma": 0.5, "Z_max": 0.03}
```

Comment:
Other parameters don't matter for a wrong SFR.
```suggestion
        model_dict = {"SFR": "InvalidModel"}
```